APIDesafioWarren/Controllers/CustomerController.cs

```diff
@@ -40,206 +39,78 @@ public IActionResult GetById(int id)
         {
             return SafeAction(() =>
             {
-                var customer = _customerServices
+                var customer = _customerAppService
                     .GetBy(x => x.Id.Equals(id));
 
-                return customer is null 
+                return customer is null
                     ? NotFound($"Customer not found! for id: {id}")
                     : Ok(customer);
             });
         }
 
-        [HttpGet("Byfullname/{fullname}")]
-        public IActionResult GetByfullname(string fullname)
+        [HttpGet("full-name/{fullname}")]
+        public IActionResult GetByFullName(string fullname)
         {
             return SafeAction(() =>
             {
-                var customer = _customerServices
+                var customer = _customerAppService
                     .GetAll(c => c.FullName == fullname);
 
-                return customer.Count is 0 
+                return customer.Count() is 0
                     ? NotFound("Customer not found!")
                     : Ok(customer);
             });
         }
 
         [HttpGet("Byemail/{email}")]
-        public IActionResult GetByemail(string email)
+        public IActionResult GetByEmail(string email)
         {
             return SafeAction(() =>
             {
-                var customer = _customerServices
+                var customer = _customerAppService
                         .GetAll(c => c.Email == email);
 
-                return customer.Count is 0 
-                    ? NotFound("Customer not found!")
-                    : Ok(customer);
-            });
-        }
-
-        [HttpGet("Bybirthdate/{birthdate}")]
-        public IActionResult GetBybirthdate(DateTime birthdate)
-        {
-            return SafeAction(() =>
-            {
-                var customer = _customerServices
-                   .GetAll(c => c.Birthdate == birthdate);
-
-                return customer.Count is 0 
+                return customer.Count() is 0
                     ? NotFound("Customer not found!")
                     : Ok(customer);
             });
         }
 
         [HttpGet("Bycpf/{cpf}")]
-        public IActionResult GetBycpf(string cpf)
+        public IActionResult GetByCpf(string cpf)
         {
             return SafeAction(() =>
             {
-                var customer = _customerServices
+                var customer = _customerAppService
                     .GetAll(c => c.Equals(cpf));
```

Comment:
```suggestion
                    .GetBy(c => c.Cpf.Equals(cpf));
```
Assim vai funcionar, pois está informando por qual campo você quer procurar o customer. Usar o GetBy porque retorna apenas um customer, pois você fez um método que impede que mais de um usuário tenha o mesmo CPF.